core-js/src/main/javascript/cdf/queries/LegacyQuery.js

```diff
@@ -53,7 +53,7 @@ define([
     interfaces: {
       lastResultSet: {
         reader: function(json) {
-          json = eval("(" + json + ")");
+          json = JSON.parse(json);
```

Comment:
For this, we might need some real unit tests to check if this is failing. For this, we must pick real query results and use them as base